Dev es servers

odscjames opened the pull request.

jarofgreen committed:
elastic: Take connection settings from ENV

jarofgreen committed:
harvesterstate: Store Harvester State  information in Elasic not SQLite

michaelwood commented on src/lib/activity-store.js:
```diff
@@ -4,10 +4,14 @@ import Elasticsearch from '@elastic/elasticsearch';
-    this.user = "test";
-    this.pass = "test";
+    this.user = process.env.ELASTICSEARCH_USERNAME || "";
+    this.pass = process.env.ELASTICSEARCH_PASSWORD || "";
```
the value doesn't need a default, the property not existing will evaluate to "undefined" which is falsey

odscjames replied:
ta

michaelwood commented on src/lib/activity-store.js:
```diff
@@ -4,10 +4,14 @@ import Elasticsearch from '@elastic/elasticsearch';
     this.esIndex = esIndex;
-    this.client = new Elasticsearch.Client({ node: 'http://localhost:9200' });
+    let client_options = { node: process.env.ELASTICSEARCH_URL || 'http://localhost:9200' };
+    if (this.user || this.pass) {
```
&& , rather than || I would have thought you'd always need both present for auth

odscjames replied:
dunno if you are allowed blank passwords but sure, let's say it should be

michaelwood commented on src/lib/activity-store.js:
```diff
@@ -4,10 +4,14 @@ import Elasticsearch from '@elastic/elasticsearch';
+    this.pass = process.env.ELASTICSEARCH_PASSWORD || "";
     this.esIndex = esIndex;
-    this.client = new Elasticsearch.Client({ node: 'http://localhost:9200' });
+    let client_options = { node: process.env.ELASTICSEARCH_URL || 'http://localhost:9200' };
```
clientOptions 
`const` rather than `let` as  we don't want the options modified afterwards. Minor thing - not sure we need to do this as a separate variable as you could pass in on the constructor as previously?

odscjames replied:
I didn't want to add the "auth" bit to the options if there wasn't any (hence let and seperate var) but for all I know elastic is totally cool with that. I'll leave this if that's ok - very minory to fix later anyway

michaelwood commented on src/lib/activity-store.js:
```diff
@@ -3,10 +3,11 @@ import Elasticsearch from '@elastic/elasticsearch';
     this.user = process.env.ELASTICSEARCH_USERNAME || "";
     this.pass = process.env.ELASTICSEARCH_PASSWORD || "";
     this.esIndex = esIndex;
+    this.esIndexMeta = esIndexMeta;
```
Could we call it esHavesterStateIndex or something like that, just because it might be confused with OA metadata

odscjames replied:
ok

jarofgreen committed:
elasticdatastructures: The document ID's in Elastic should be a mix of publisher ID and Data ID

Otherwise the same data ID from different publishers will collide.

https://www.w3.org/2017/08/realtime-paged-data-exchange/#-item- says for ID's: "this specification does not prescribe any specific format."

jarofgreen committed:
elastic: catch 404 on delete and index without deleting

Catch errors on delete. It's sometimes expected to get a 404.
we may be seeing a delete message from the API for a data item we have never seen before

We don't need to delete before doing an index - index will replace too.

Also added a TODO to a code coment.
